make bazel-build fails with `ln: failed to create symbolic link '/tmp/gopath/src/k8s.io/kubernetes': File exists`

Running `make bazel-build` (on master), gives the error `ln: failed to create symbolic link '/tmp/gopath/src/k8s.io/kubernetes': File exists`:


```
export GOPATH=/tmp/gopath
export GO_WORKSPACE=${GOPATH}/src/k8s.io/kubernetes
mkdir -p ${GO_WORKSPACE%/*}
ln -s $(pwd)/ ${GO_WORKSPACE}
if [[ ! -e ${GO_WORKSPACE}/external ]]; then ln -s $(p...' failed: bash failed: error executing command 
  (exec env - \
    GOARCH=amd64 \
    GOOS=linux \
    GOROOT=/home/justinsb/.cache/bazel/_bazel_justinsb/53419f5d51785321b16f924d43766350/external/go1.8.3.linux-amd64 \
  /bin/bash -c 'set -e
export GOPATH=/tmp/gopath
export GO_WORKSPACE=${GOPATH}/src/k8s.io/kubernetes
mkdir -p ${GO_WORKSPACE%/*}
ln -s $(pwd)/ ${GO_WORKSPACE}
if [[ ! -e ${GO_WORKSPACE}/external ]]; then ln -s $(pwd)/external/ ${GO_WORKSPACE}/; fi
if [[ ! -e ${GO_WORKSPACE}/bazel-out ]]; then ln -s $(pwd)/bazel-out/ ${GO_WORKSPACE}/; fi
export GENGOPATH=/tmp/gengopath
export GENGO_WORKSPACE=${GENGOPATH}/src/k8s.io/kubernetes
mkdir -p ${GENGO_WORKSPACE%/*}
ln -s $(pwd)/bazel-out/local-fastbuild/genfiles ${GENGO_WORKSPACE}
export GOPATH=${GOPATH}:${GENGOPATH}
cd ${GO_WORKSPACE}
bazel-out/host/bin/vendor/k8s.io/kube-gen/cmd/set-gen/set-gen     --input-dirs ./vendor/k8s.io/apimachinery/pkg/util/sets/types     --output-base $(dirname $(dirname bazel-out/local-fastbuild/genfiles/vendor/k8s.io/apimachinery/pkg/util/sets/byte.go))     --go-header-file hack/boilerplate/boilerplate.go.txt     --output-package sets')

Use --sandbox_debug to see verbose messages from the sandbox.
ln: failed to create symbolic link '/tmp/gopath/src/k8s.io/kubernetes': File exists
Use --strategy=GoGenrule=standalone to disable sandboxing for the failing actions.
INFO: Elapsed time: 3.383s, Critical Path: 0.65s
Makefile:527: recipe for target 'bazel-build' failed
make: *** [bazel-build] Error 1
```


Bazel version 0.5.2:
```
Build label: 0.5.2
Build target: bazel-out/local-fastbuild/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Tue Jun 27 13:27:03 2017 (1498570023)
Build timestamp: 1498570023
Build timestamp as int: 1498570023
```

cc @kubernetes/sig-release-bugs

--- a/build/root/WORKSPACE
+++ b/build/root/WORKSPACE
@@ -7,9 +7,9 @@ http_archive(
 
 http_archive(
     name = "io_kubernetes_build",
-    sha256 = "2b2cb64b2fd7734c5eb2b79b79fa35d064fcf53b92cb3aaec5b90fc10fc94135",
-    strip_prefix = "repo-infra-9ba3a24eeffafa3a794b9e7312103424e7da26e9",
-    urls = ["https://github.com/kubernetes/repo-infra/archive/9ba3a24eeffafa3a794b9e7312103424e7da26e9.tar.gz"],
+    sha256 = "5da78568ffb9a323410c701618c23da8c93f4bf4aea76eee41ac244dbd8c8f95",
+    strip_prefix = "repo-infra-4eaf9e671bbb549fb4ec292cf251f921d7ef80ac",
+    urls = ["https://github.com/kubernetes/repo-infra/archive/4eaf9e671bbb549fb4ec292cf251f921d7ef80ac.tar.gz"],
 )
 
 ETCD_VERSION = "3.0.17"

--- a/hack/update-bazel.sh
+++ b/hack/update-bazel.sh
@@ -25,7 +25,7 @@ source "${KUBE_ROOT}/hack/lib/init.sh"
 rm -f "${KUBE_ROOT}/pkg/generated/openapi/zz_generated.openapi.go"
 
 # The git commit sha1s here should match the values in $KUBE_ROOT/WORKSPACE.
-kube::util::go_install_from_commit github.com/kubernetes/repo-infra/kazel 9ba3a24eeffafa3a794b9e7312103424e7da26e9
+kube::util::go_install_from_commit github.com/kubernetes/repo-infra/kazel 4eaf9e671bbb549fb4ec292cf251f921d7ef80ac
 kube::util::go_install_from_commit github.com/bazelbuild/rules_go/go/tools/gazelle/gazelle 82483596ec203eb9c1849937636f4cbed83733eb
 
 gazelle fix -build_file_name=BUILD,BUILD.bazel -external=vendored -mode=fix -repo_root="$(kube::realpath ${KUBE_ROOT})"

--- a/hack/verify-bazel.sh
+++ b/hack/verify-bazel.sh
@@ -25,7 +25,7 @@ source "${KUBE_ROOT}/hack/lib/init.sh"
 rm -f "${KUBE_ROOT}/pkg/generated/openapi/zz_generated.openapi.go"
 
 # The git commit sha1s here should match the values in $KUBE_ROOT/WORKSPACE.
-kube::util::go_install_from_commit github.com/kubernetes/repo-infra/kazel 9ba3a24eeffafa3a794b9e7312103424e7da26e9
+kube::util::go_install_from_commit github.com/kubernetes/repo-infra/kazel 4eaf9e671bbb549fb4ec292cf251f921d7ef80ac
 kube::util::go_install_from_commit github.com/bazelbuild/rules_go/go/tools/gazelle/gazelle 82483596ec203eb9c1849937636f4cbed83733eb
 
 gazelle_diff=$(gazelle fix -build_file_name=BUILD,BUILD.bazel -external=vendored -mode=diff -repo_root="$(kube::realpath ${KUBE_ROOT})")

